sdk/servicebus/service-bus/src/log.ts

```diff
@@ -82,6 +85,14 @@ export interface ServiceBusLogger extends AzureLogger {
   logError(err: Error | AmqpError | undefined, ...args: any[]): void;
 }
 
+function isRetryableError(error: any): error is AmqpError {
+  if (!rheaIsAmqpError(error) && !isError(error)) {
+    return false;
+  }
```

Comment:
`rheaIsAmqpError` is already called by `translate` in https://github.com/Azure/azure-sdk-for-js/blob/2388847cce37000dbb4dcc65b699019cca903655/sdk/core/core-amqp/src/errors.ts#L687. Is there a reason to call it earlier here?

Reply:
It's likely from previous iteration of my code and should be removed now.

Reply:
nitpick: We could also skip the `isError` check and let `translate` handle it: https://github.com/Azure/azure-sdk-for-js/blob/2388847cce37000dbb4dcc65b699019cca903655/sdk/core/core-amqp/src/errors.ts#L755 but I understand if we want to short circuit early. Could we perhaps teach `translate` to short circuit early?

Reply:
thinking about it more. maybe I should extract out `isRetryable` and use it in here and translate so we don't have do translation when not needed